fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.kt

```diff
@@ -190,13 +217,27 @@ class SiteRestClient @Inject constructor(
     ): NewSiteResponsePayload {
         val url = WPCOMREST.sites.new_.urlV1_1
         val body = mutableMapOf<String, Any>()
-        body["blog_name"] = siteName
         body["lang_id"] = language
         body["public"] = visibility.value().toString()
         body["validate"] = if (dryRun) "1" else "0"
         body["client_id"] = appSecrets.appId
         body["client_secret"] = appSecrets.appSecret
 
+        if (siteTitle != null) {
+            body["blog_title"] = siteTitle
+        }
+        if (siteName != null) {
+            body["blog_name"] = siteName
+        } else {
+            if (siteTitle != null) {
+                body["blog_name"] = siteTitle
+            } else {
+                body["blog_name"] = username
+            }
+            body["site_creation_flow"] = "with-design-picker"
+            body["find_available_url"] = "1"
+        }
```

Comment:
We could simplify this (removing the nested `if`s) with:

```suggestion
        body["blog_name"] = siteName ?: siteTitle ?: username
        siteName ?: run {
            body["site_creation_flow"] = "with-design-picker"
            body["find_available_url"] = "1"
        }
```

Or, alternatively, with keeping one of the `if`s:

```suggestion
        if (siteName != null) {
            body["blog_name"] = siteName
        } else {
            body["blog_name"] = siteTitle ?: username
            body["site_creation_flow"] = "with-design-picker"
            body["find_available_url"] = "1"
        }
```

Wdyt?

Reply:
Great suggestions @mkevins 🙇 
I liked more and merged the 1st one :)